### PR TITLE
Upgrade website netlify hugo version

### PR DESCRIPTION
**Pull request by mrcdb**
This PR attempts to bump the Hugo version used by netlify when building the website from 0.115.2 (released in [July 2023](https://github.com/gohugoio/hugo/releases/tag/v0.115.2)) to 0.128.0, the [latest stable release at this time](https://github.com/gohugoio/hugo/releases/tag/v0.128.0). 

The PR also bumps the hugo-extended npm package version to latest stable: https://www.npmjs.com/package/hugo-extended/v/0.127.0

### CODE REVIEW

**Comment by mrcdb** [comment]
It's worth mentioning that the netlify build logs show a couple of deprecation warnings with the newer version:

```text
7:35:02 AM: WARN  deprecated: .Site.IsServer was deprecated in Hugo v0.120.0 and will be removed in a future release. Use hugo.IsServer instead.
7:35:03 AM: WARN  deprecated: .Site.GoogleAnalytics was deprecated in Hugo v0.120.0 and will be removed in a future release. Use .Site.Config.Services.GoogleAnalytics.ID instead.
```

**Comment by eddie-knight** [comment]
Thanks @mrcdb ✊ 

I don't see any issues or anticipate difficulty from this, but I would consider a version bump to be a potentially destructive activity... So let's wait until other website PRs merge and keep this open at minimum 24 hours for additional feedback.

**Review by jkjell — APPROVED** [review]

**Pull request merged**
